stingray/largememory.py

```diff
@@ -0,0 +1,558 @@
+                                           mode='r',
+                                           path='energy')
+        except ValueError:
+            energy_other = None
+
+        try:
+            gti_other = zarr.open_array(store=data_path[2],
+                                        mode='r',
+                                        path='gti')
+        except ValueError:
+            gti_other = None
+
+        try:
+            pi_channel_other = zarr.open_array(store=data_path[2],
+                                               mode='r',
+                                               path='pi_channel')
+        except ValueError:
+            pi_channel_other = None
+
+        fin_spec = stingray.AveragedCrossspectrum()
+
+    else:
+        raise ValueError
+
+    flag = True
+    for i in range(times.chunks[0], times.size, times.chunks[0]):
+        ev1 = EventList(
+            time=times.get_basic_selection(slice(i - times.chunks[0], i)),
+            energy=energy.get_basic_selection(slice(i - times.chunks[0], i))
+            if energy is not None else None,
+            ncounts=ncounts[...],
+            mjdref=mjdref[...],
+            dt=dt[...],
+            gti=gti[...] if gti is not None else None,
+            pi=pi_channel[...] if pi_channel is not None else None)
+
+        if spec_type == 'AveragedPowerspectrum':
+            if segment_size < ev1.time.size / 8192:
+                warnings.warn(
+                    f"It is advisable to have the segment size greater than or equal to {ev1.time.size / 8192}. Very small segment sizes may greatly increase computation times."
+                )
+
+            avg_pspec = stingray.AveragedPowerspectrum(data=ev1, segment_size=ev1.time.size / segment_size, norm=norm, gti=gti, silent=silent, dt=dt1, large_data=False)
+
+            fin_spec = _addSpectra(fin_spec, avg_pspec, flag)
+
+        elif spec_type == 'AveragedCrossspectrum':
+            ev2 = EventList(time=times_other.get_basic_selection(slice(i - times.chunks[0], i)), energy=energy_other.get_basic_selection(slice(i - times.chunks[0], i) if energy_other is not None else None, ncounts=ncounts_other[...], mjdref=mjdref_other[...], dt=dt_other[...], gti=gti_other[...] if gti_other is not None else None, pi=pi_channel_other[...] if pi_channel_other is not None else None))
+
+            if segment_size < ev1.time.size / 4096:
+                warnings.warn(
+                    f"It is advisable to have the segment size greater than or equal to {ev1.time.size / 4096}. Very small segment sizes may greatly increase computation times."
+                )
+
+            avg_cspec = stingray.AveragedCrossspectrum(data1=ev1, data2=ev2, segment_size=ev1.time.size / segment_size, norm=norm, gti=gti, power_type=power_type, silent=silent, dt=dt1, large_data=False)
+
+            fin_spec = _addSpectra(fin_spec, avg_cspec, flag)
+
+        flag = False
+
+    return fin_spec
+
+
+def saveData(data_obj, f_name):
```

Comment:
Together with the saveData function, there should be a retrieveData or similar that reads the data back into the correct Eventlist or Lightcurve objects.

Reply:
Okay will do! But will it read back the whole data into the EventList/Lightcurve or just chunks of it?

Reply:
We can make a filter to only get a part of the dataset, it would be very good. But also retrieving the full dataset would work for medium-size data.

Reply:
I will implement that